CHANGELOG.md

```diff
@@ -2,6 +2,114 @@
+    * using a worker thread.
+    *
+    * If using runtime styling:
+    *
+    * loadStyle(style(Style.DARK) {
+    *   +geoJsonSource(id) {
+    *    featureCollection(collection)
+    *   }
+    *   ...
+    * }
+    *
+    * compositing style will be performed correctly under the hood and
+    * [Style.OnStyleLoaded] will be emitted in correct moment of time when all sources are parsed.
+    *
+    * If creating geojson sources for already loaded Style please consider using overloaded
+    * geoJsonSource(String, GeoJsonSource.Builder.() -> Unit, onGeoJsonParsed: (GeoJsonSource) -> Unit) function
+    * and use fully prepared [GeoJsonSource] in onGeoJsonParsed callback.
+    */
+   fun geoJsonSource(
+     id: String,
+     block: GeoJsonSource.Builder.() -> Unit
+   ): GeoJsonSource
+   
+   /**
+    * DSL function for [GeoJsonSource] performing parsing using a worker thread.
+    * Immediately returns [GeoJsonSource] with no data set,
+    * fully parsed [GeoJsonSource] is returned in [onGeoJsonParsed] callback.
+    *
+    * Using this method means that it is user's responsibility to proceed with adding this source,
+    * layers or other style objects in [onGeoJsonParsed] callback.
+    */
+   fun geoJsonSource(
+     id: String,
+     block: GeoJsonSource.Builder.() -> Unit,
+     onGeoJsonParsed: (GeoJsonSource) -> Unit
+   ): GeoJsonSource
+   ```
+   
+   If updating existing `GeoJsonSource` from some `Style` overloaded functions are introduced:
+   
+   ```
+   /**
+      * Add a FeatureCollection to the GeojsonSource.
+      * If [onDataParsed] is provided and not null - data will be loaded in async mode.
+      * Otherwise method will be synchronous.
+      *
+      * @param value the feature collection
+      * @param onDataParsed optional callback notifying when data is parsed on a worker thread
+      */
+     fun featureCollection(
+       value: FeatureCollection,
+       onDataParsed: ((GeoJsonSource) -> Unit)? = null
+     )
+   ```
+   Similar functions are introduced for `feature` and `geometry`.
+
+## Features ✨ and improvements 🏁
+* Add a `cameraOptions(cameraState, builderBlock)` inline method that helps mutate an existing `CameraState` object. ([#317](https://github.com/mapbox/mapbox-maps-android/pull/317))
+* Add selected state handling to annotation plugin ([#316](https://github.com/mapbox/mapbox-maps-android/pull/316))
+* Add API for disabling vertical/horizontal scroll gestures ([#319](https://github.com/mapbox/mapbox-maps-android/pull/319))
+* Introduce API to enable render cache feature that could bring up rendering performance improvement. ([#326](https://github.com/mapbox/mapbox-maps-android/pull/326))
+* Add `removeAnnotationManager` API. ([#330](https://github.com/mapbox/mapbox-maps-android/pull/330))
+* Improve terrain's rendering performance 
+* Set 'begin' and 'end' timestamps for StyleLoaded and MapLoaded events, so that developers could check how much time it takes to load style and map, respectively
```

Comment:
```suggestion
* Set `begin` and `end` timestamps for StyleLoaded and MapLoaded events, so that developers could check how much time it takes to load style and map, respectively
```